source/opt/ir_loader.cpp

```diff
@@ -87,7 +87,7 @@ void IrLoader::AddInstruction(const spv_parsed_instruction_t* inst) {
         module_->AddType(std::move(spv_inst));
       } else if (IsConstantInst(opcode)) {
         module_->AddConstant(std::move(spv_inst));
-      } else if (opcode == SpvOpVariable) {
+      } else if (opcode == SpvOpVariable || opcode == SpvOpUndef) {
```

Comment:
What about only having `AddType` and `AddGlobalValue`? We can use the latter to add constants, variables, & `OpUndef`s.